### PR TITLE
markdown toolbar

### PR DESCRIPTION
**Pull request by keithamus**
Closes https://github.com/github/primer/issues/3107

Adds a new `MarkdownToolbar` component leveraging `<markdown-toolbar>` to allow for a `<textarea>` to take formatting buttons.

### Changelog

- Adds a new MarkdownToolbar draft component.

#### New

 - MarkdownToolbar Compoent

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

**Review by camertron — APPROVED** [review]
Nice! 🎉

File: generated/components.json
Comment (owenniblock) [inline generated/components.json:1]: Wondering what's going on with this file 🤔

File: packages/react/package.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
   },
   "dependencies": {
     "@github/combobox-nav": "^2.1.5",
-    "@github/markdown-toolbar-element": "^2.1.0",
+    "@github/markdown-toolbar-element": "2.2.3",
```
Comment (owenniblock) [inline packages/react/package.json:88]: Do we want to copy the pattern of other deps here?

```suggestion
    "@github/markdown-toolbar-element": "^2.2.3",
```

File: packages/react/src/drafts/MarkdownToolbar/MarkdownToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  StrikethroughIcon,
+  TasklistIcon,
+} from '@primer/octicons-react'
+import MarkdownToolbarElement from '@github/markdown-toolbar-element'
+
+const MarkdownToolbarComponent = createComponent(MarkdownToolbarElement, 'markdown-toolbar')
+MarkdownToolbarComponent.displayName = 'MarkdownToolbar'
+
+export interface MarkdownToolbarProps extends ComponentProps<typeof MarkdownToolbarComponent> {
+  for?: string
```
Comment (owenniblock) [inline packages/react/src/drafts/MarkdownToolbar/MarkdownToolbar.tsx:29]: Should this be required?

**Review by owenniblock — APPROVED** [review]
Looks great! A couple of thoughts / questions but non-blocking.

File: generated/components.json
Comment (keithamus) [reply generated/components.json:1]: Me too :confused:

**Comment by maximedegreve** [comment]
This is a great addition! There are a few things I've spotted while reviewing this work in Storybook.

- While not directly part of this component I've noticed that in Safari the <TextArea> isn't styled.
- The padding seems to be missing in the parent of the buttons.
- There should be no spacing between the buttons

Showing before and after

**Review by mperrotti — COMMENTED** [review]
The issue this resolves (https://github.com/github/primer/issues/3107) mentions a "CommentEditor". What is that?

I'm wondering if we really want this in PRC since we've deprecated similar components:
- MarkdownEditor
- MarkdownViewer

**Comment by keithamus** [comment]
> The issue this resolves ([github/primer#3107](https://github.com/github/primer/issues/3107)) mentions a "CommentEditor". What is that?

That is the comment editing experience in dotcom. I'm happy to not include this in PRC and just keep it to dotcom.

**Comment by keithamus** [comment]
We'll keep this in the monolith.

**Pull request closed without merging**